programmers/4일차/로또의-최고-순위와-최저-순위.js

```diff
@@ -0,0 +1,36 @@
+function solution(lottos,winNums){
+    const correctCntRank = {
+        6: 1,
+        5: 2,
+        4: 3,
+        3: 4,
+        2: 5,
+        1: 6,
+        0: 6
+    }
+    let minCount = 0;
+    let zeroCount = 0;
+    let maxCount = 0;
+    //minCount,maxCount 은 로또의 최소,최고 등수일때 zeroCount 은 lottos에 있는 0의 개수
+    
+    for (let zero of lottos) {
+        if (zero == 0) zeroCount += 1;
+    }
+    // console.log(zeroCount)
+    for (let num of lottos) {
+        if (winNums.includes(num)) {
+            minCount += 1;
+        }
+    } 
+    // console.log(minCount)
+    maxCount = minCount + zeroCount
+    return [correctCntRank[maxCount],correctCntRank[minCount]]
+}
```

Comment:
```suggestion
function solution(lottos, win_nums) {
   const rank = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6};
   const answer = [];
   let winCount = 0;
   let zeroCount = 0;

   winCount = win_nums.filter((num) => lottos.includes(num)).length;
   zeroCount = lottos.filter((num) => num === 0).length;

   return [rank[winCount + zeroCount], rank[winCount]];
}
```
정말 잘 푸셨네요  제 풀이도  접근법이  같아요  하나 알려드리고 싶은 것은  filter메서드를 이용하면  for문을 
이용하지 않고도  더 간결하게 풀이가 가능합니다  사실  for문을 정말 써야 되는 상황들도 있지만  대부분 
forEach,  map, filter, reduce 같은 메서드들로 해결이 가능해요  효율성이나  코드의 간결함에 있어서도 더 좋구요
이 4가지 메서드들  코테에서도 자주 사용하게 되는 메서드이기 때문에  한번 공부해보시는것을 추천드립니다 
문제 정말 잘 푸셨어요  컨벤션도 피드백 드릴게 없네요 ㅎㅎㅎ